src/Locations/Entities/CustomOpeninghours.php

```diff
@@ -244,6 +262,57 @@ protected function getOpeningHours(\DateTime $date)
         return end($timeslot) ?? $timeslot;
     }
 
+    /**
+     * Get first upcoming open timeslot of today.
+     */
+    protected function getUpcomingOpenTimeslotToday(\DateTime $time): ?Timeslot
```

Comment:
Liever geen null gebruiken als het niet hoeft:

https://hackernoon.com/null-the-billion-dollar-mistake-8t5z32d6
https://dzone.com/articles/the-worst-mistake-of-computer-science-1

Je kan ook false returnen  `false|Timeslot`.

Maar het liefst strict type toepassen (een functie geeft altijd 1 type terug).

Je kan een extra functie maken als `upcomingOpenTimeslotTodayExists` of `hasUpcomingOpenTimeslotToday`. Iets meer werk maar veel beter te lezen en strictly typed :)

Je kan ook all the way gaan met een "Null object pattern": https://front-line-php.com/dealing-with-null
Maar dat is wellicht overkill :)